libraries/chain/db_management.cpp

```diff
@@ -46,27 +46,36 @@ database::~database()
    clear_pending();
 }
 
-void database::reindex(fc::path data_dir, const genesis_state_type& initial_allocation)
+void database::reindex( fc::path data_dir )
 { try {
-   ilog( "reindexing blockchain" );
-   wipe(data_dir, false);
-   open(data_dir, [&initial_allocation]{return initial_allocation;});
-
-   auto start = fc::time_point::now();
    auto last_block = _block_id_to_block.last();
    if( !last_block ) {
       elog( "!no last block" );
       edump((last_block));
       return;
    }
+   if( last_block->block_num() <= head_block_num()) return;
 
+   ilog( "reindexing blockchain" );
+   auto start = fc::time_point::now();
    const auto last_block_num = last_block->block_num();
+   uint32_t flush_point = last_block_num - 10000;
+   uint32_t undo_point = last_block_num - 50;
 
```

Comment:
Good point, but doesn't matter in practice IMO.
If last_block_num < 10000 then flush_point > MAX_INT-10000, which means we never flush. We don't lose much though, only the state from less than 10k blocks.
If last_block_num < 50 then undo_point > MAX_INT - 50, which means the fork_database isn't used during replay. But that is only a problem if your block_database ends in a fork, which is unlikely.